src/test/java/com/pgs/client/supplier/AccessTokenSupplierTest.java

```diff
@@ -0,0 +1,96 @@
+package com.pgs.client.supplier;
+
+import com.pgs.client.dto.Token;
+import com.pgs.client.service.AuthenticationClient;
+import lombok.SneakyThrows;
+import org.apache.commons.lang3.time.StopWatch;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class AccessTokenSupplierTest {
+
+    @Mock
+    private AuthenticationClient authenticationClient;
+    @InjectMocks
+    private AccessTokenSupplier accessTokenSupplier;
+    @Mock
+    private StopWatch stopWatch;
+
+    private static final Token TOKEN1 = Token.builder()
+            .accessToken("access1234")
+            .refreshToken("refresh1234")
+            .build();
+    private static final Token TOKEN2 = Token.builder()
+            .accessToken("access4321")
+            .refreshToken("refresh4321")
+            .build();
+
+    @Test
+    void testSupplyToken() {
+        ReflectionTestUtils.setField(accessTokenSupplier, "ttl", Duration.ofHours(1));
+        when(authenticationClient.getToken())
+                .thenReturn(TOKEN1);
+        var accessTokenString = accessTokenSupplier.supplyToken();
+        verify(authenticationClient).getToken();
+        assertEquals(accessTokenString, TOKEN1.getAccessToken());
+
+    }
+
+    @SneakyThrows
+    @Test
+    void testNewTokenSupplyPastTTL() {
+        String accessTokenString1 = "";
+        String accessTokenString2 = "";
+        when(authenticationClient.getToken())
+                .thenReturn(TOKEN1).thenReturn(TOKEN2);
+        stopWatch.start();
```

Comment:
in this method you should have inject the mock stopwatch (using ReflectionTestUtils) into supplier and mock it's behavior rather than calling it's methods